Mathlib/Algebra/Module/Submodule/Map.lean

```diff
@@ -619,3 +619,25 @@ def compatibleMaps : Submodule R (N →ₗ[R] N₂) where
 #align submodule.compatible_maps Submodule.compatibleMaps
 
 end Submodule
+
+namespace LinearMap
+
+variable { M M' R : Type* } [Semiring R] [AddCommMonoid M] [AddCommMonoid M']
+  [Module R M] [Module R M']
+
+/-- A linear map between two modules restricts to a linear map from any submodule p of the
+domain onto the image of that submodule.
+
+This is the linear version of AddMonoidHom.addSubmonoidMap and AddMonoidHom.addSubgroupMap.-/
+def submoduleMap (f : M →ₗ[R] M') (S : Submodule R M) : S →ₗ[R] S.map f where
+  toFun := fun x ↦ ⟨ f x, Submodule.apply_coe_mem_map f x ⟩
```

Comment:
Note that [LinearEquiv.submoduleMap](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Basic.html#LinearEquiv.submoduleMap) is defined using domRestrict and codRestrict.